include/dsn/dist/replication/replication_app_client_base.h

```diff
@@ -96,7 +106,7 @@ namespace dsn { namespace replication {
         
         template<typename TRequest, typename TResponse>
         ::dsn::task_ptr write(
-            int partition_index,
+            uint32_t key_hash,
```

Comment:
better to use uint64_t for hash to have larger space to play for the upper applications.